src/Common/WKB.cpp

```diff
@@ -0,0 +1,119 @@
+#include "Common/Exception.h"
+#include <Common/WKB.h>
+#include "Functions/geometryConverters.h"
+
+#include <IO/ReadBuffer.h>
+#include <IO/ReadHelpers.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+}
+
+inline CartesianPoint readPointWKB(ReadBuffer & in_buffer, std::endian endian_to_read)
+{
+    Float64 x;
+    Float64 y;
+    readBinaryEndian(x, in_buffer, endian_to_read);
+    readBinaryEndian(y, in_buffer, endian_to_read);
+    return CartesianPoint(x, y);
+}
+
+inline LineString<CartesianPoint> readLineWKB(ReadBuffer & in_buffer, std::endian endian_to_read)
+{
+    Int32 num_points;
+    readBinaryEndian(num_points, in_buffer, endian_to_read);
+
+    LineString<CartesianPoint> line;
+    for (int i = 0; i < num_points; ++i)
+    {
+        line.push_back(readPointWKB(in_buffer, endian_to_read));
+    }
+    return line;
+}
+
+inline Polygon<CartesianPoint> readPolygonWKB(ReadBuffer & in_buffer, std::endian endian_to_read)
+{
+    Int32 num_lines;
+    readBinaryEndian(num_lines, in_buffer, endian_to_read);
+
+    Polygon<CartesianPoint> polygon;
+    for (int i = 0; i < num_lines; ++i)
+    {
+        auto parsed_points = readLineWKB(in_buffer, endian_to_read);
+        if (i == 0)
+        {
+            polygon.outer().reserve(parsed_points.size());
```

Comment:
We might want to do validation for rings (number of points, first and last point is the same), but we can add it as an option/setting later on.